src/modules/dataman/dataman.c

```diff
@@ -949,7 +949,15 @@ _file_initialize(unsigned max_offset)
 		close(dm_operations_data.file.fd);
 
 		if (incompat) {
+#ifdef CONFIG_ARCH_BOARD_AEROCORE2
```

Comment:
This last one fails on a px4fmu-v2 as well.  I turned off autostart, at the console:
nsh>mtd start
nsh>mtd erase
Erasing /fs/mtd_params
Erased 8192 bytes
Erasing /fs/mtd_waypoints
Erased 8192 bytes
nsh>dataman start -f /fs/mtd_waypoints
INFO  [dataman] dataman file set to: /fs/mtd_waypoints
WARN  [dataman] Could not open data manager file /fs/mtd_waypoints
ERROR [dataman] dataman start failed

if you check the file system
nsh>ls /fs
/fs:
 bbr0
 bbr1
 bbr2
 mtd_params

It's missing mtd_waypoints.  This wasn't the case with the stable branch in January (not sure about now).  It was the case for the master branch in January, I didn't have time to investigate then.  For whatever reason, "unlink" is removing /fs/mtd_waypoints which is causing the "open" command to fail.

I'm investigating ...